SIPS/sip-26.md

```diff
@@ -0,0 +1,163 @@
+---
+sip: 26
+title: Account Router Snaps
+status: Draft
+author: Daniel Rocha (@danroc), Frederik Bolding (@FrederikBolding), Alex Donesky (@adonesky1)
+created: 2024-08-28
+---
+
+## Abstract
+
+This SIP presents an architecture to enable Snaps to expose blockchain-specific
+methods to dapps, extending MetaMask's functionality to support a multichain
+ecosystem.
+
+## Motivation
+
+Currently, MetaMask is limited to EVM-compatible networks. This proposal aims
+to empower developers, both first- and third-party, to use Snaps to add native
+support for non-EVM-compatible chains within MetaMask.
+
+## Specification
+
+> Formal specifications are written in TypeScript.
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written
+in uppercase in this document are to be interpreted as described in [RFC
+2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### High-Level architecture
+
+The diagram below represents a high-level architecture of how the RPC Router
+integrates inside MetaMask to allow Snaps to expose protocol methods to dapps
+and the MetaMask clients.
+
+![High-level architecture](../assets/sip-26/components-diagram.png)
+
+- **Account Snaps**: Snaps that implement the Keyring API and are responsible
+  for signing requests and managing accounts.
+
+- **Protocol Snaps**: Snaps that implement protocol methods that do not require
+  an account to be executed.
+
+- **RPC Router**: Native component that forwards RPC requests to the
+  appropriate Protocol Snap, Account Snap or native implementation.
```

Comment:
I believe its more likely that requests that can be handled by native implementation will not reach the RPC Router
```suggestion
  appropriate Protocol Snap or Account Snap.
```

Reply:
This also appears to match @FrederikBolding's [current implementation](https://github.com/MetaMask/snaps/pull/2875/files#diff-5697fc531248874f2ce02e26e9e13fee8222753070a8c8e107de6c281ba220afR246)

Reply:
Resolved this in my PR